### PR TITLE
Links: Filter datasources in derived link to tracing only

### PR DESCRIPTION
**Pull request by aocenas**
Part of fix for this issue https://github.com/grafana/grafana/issues/23935.

Basically only tracing datasources work right now with internal links so this allow just those in the datasource dropdown.

### CODE REVIEW

File: docs/sources/features/datasources/loki.md [hunk 1]
```diff
@@ -189,14 +189,15 @@ datasources:
     jsonData:
       maxLines: 1000
       derivedFields:
-        # Field with internal link pointing to datasource in Grafana
+        # Field with internal link pointing to datasource in Grafana.
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/loki.md:192]:
```suggestion
        # Field with internal link pointing to data source in Grafana.
```

Comment (oddlittlebird) [reply docs/sources/features/datasources/loki.md:192]: Per the [Documentation style guide](https://github.com/grafana/grafana/blob/master/contribute/style-guides/documentation-style-guide.md), "data source" is two words.

File: docs/sources/features/datasources/loki.md [hunk 2]
```diff
@@ -189,14 +189,15 @@ datasources:
...
+        # At this moment only Jaeger and Zipkin datasource is supported as the link target.
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/loki.md:193]:
```suggestion
        # Right now, Grafana supports only Jaeger and Zipkin data sources as link targets.
```

**Review by oddlittlebird — DISMISSED** [review]

**Comment by oddlittlebird** [comment]
Doc changes look good, approved.

**Review by davkal — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/loki/configuration/DerivedField.tsx [hunk 3]
```diff
@@ -152,6 +152,8 @@ const DataSourceSection = (props: DataSourceSectionProps) => {
   const { datasourceUid, onChange } = props;
   const datasources: DataSourceSelectItem[] = getDatasourceSrv()
     .getExternal()
+    // At this point internal link do not work with any other datasource than Jaeger and Zipkin
```
Comment (davkal) [inline public/app/plugins/datasource/loki/configuration/DerivedField.tsx:155]: I'd prolly written this similarly positive as the comment in the other file:
```
# At this moment only Jaeger and Zipkin datasource is supported as the link target.
```

**Pull request merged**